jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -280,27 +280,31 @@ else if (!cookie.getPath().equals(oldCookie.getPath()))
 
     @Override
     public void addCookie(Cookie cookie)
-    {
-        if (StringUtil.isBlank(cookie.getName()))
-            throw new IllegalArgumentException("Cookie.name cannot be blank/null");
-
-        String comment = cookie.getComment();
-        // HttpOnly was supported as a comment in cookie flags before the java.net.HttpCookie implementation so need to check that
-        boolean httpOnly = cookie.isHttpOnly() || HttpCookie.isHttpOnlyInComment(comment);
-        SameSite sameSite = HttpCookie.getSameSiteFromComment(comment);
-        comment = HttpCookie.getCommentWithoutAttributes(comment);
-
-        addCookie(new HttpCookie(
-            cookie.getName(),
-            cookie.getValue(),
-            cookie.getDomain(),
-            cookie.getPath(),
-            (long)cookie.getMaxAge(),
-            httpOnly,
-            cookie.getSecure(),
-            comment,
-            cookie.getVersion(),
-            sameSite));
+    { 
+        //Servlet Spec 9.3 Include method: cannot set a cookie if handling an include
+        if (isMutable())
+        {
```

Comment:
So it silently fails if an attempt is made to addCookie during an INCLUDE?

Reply:
Can we at least log at debug level this behavior?

Such as ...

``` java
if (!isMutable())
{
    LOG.debug("Not able to add/set Cookie during INCLUDE dispatch: {}", cookie);
    return;
}

if (StringUtil.isBlank(cookie.getName()))
    throw new IllegalArgumentException("Cookie.name cannot be blank/null");

...
```

Reply:
Yep, that's what the spec wants us to do, ignore it. All of the other api methods that add headers do the same isMutable() test and ignore if an include.

Reply:
Yeah, typical strange spec behavior.
Can we (jetty) at least offer debug messages for those that are attempting to troubleshoot why their application isn't behaving like they expect?

Reply:
Don't worry about the debug thing on this PR.
I've opened issue #4916 to address this in a more consistent way.